### DISCUSSION TITLE
TOC customization changed in 2.0.0-beta.16

### DISCUSSION THREAD

**Discussion by mbj4668** [facebook/docusaurus#6962] [General] [answered]
In #6729, the old TOC splicing behavior `toc={[toc[2], toc[4]]}` was removed.

We use this function to split the TOC into a few parts, with some additional text inbetween.

Is there any way to achieve this behavior now?

**Comment by slorber** [comment, marked as answer]
Yes, the TOC was refactored from a tree structure to a flat array structure, but you can still filter it to create a slice of that TOC array

One example:

```diff
- <TOCInline toc={toc[toc.length - 1].children}/>
+ <TOCInline toc={toc.slice(2)} />
```

See also https://github.com/facebook/jest/pull/12500#issuecomment-1059037963

---

Note: if your toc transformations are complex and not easy to inline in MDX, you can build your own helper to filter this TOC. 

Eventually, we could add these utils to Docusaurus but for that we'd need much more info about your use-case and see if other users are presenting a similar use-case to design a good API

**Comment by mbj4668** [comment]
Thank you for the proposal!  This solves our problem.

We want to group the TOC into "sections" and add some text before each such "section":

```
Text for first section
  o  heading A
    o  heading A.1
    o  heading A.2
  o heading B

Text for second section
  o  heading C
...
```

We can do this with the slice-approach you suggested; It is a bit fragile, but we can live with that.
